src/ORM/FieldType/DBComposite.php

```diff
@@ -270,11 +271,11 @@ public function setField($field, $value, $markChanged = true)
     {
         $this->objCacheClear();
 
-        // Non-db fields get assigned as normal properties
         if (!$this->hasField($field)) {
-            parent::setField($field, $value);
-
-            return $this;
+            throw new InvalidArgumentException(implode(' ', [
+                "Field $field does not exist.",
+                'If this was accessed via a dynamic property then use ViewableData::setDynamicData() instead.'
```

Comment:
```suggestion
                'If this was accessed via a dynamic property then use setDynamicData() instead.'
```
We don't call `ViewableData::setDynamicData('myProperty')`, we just call `setDynamicData` on the instance of `DBComposite`.

Reply:
Updated.  Also added unit test for the exception